src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.companieshouse.pscdataapi.transform;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.psc.FullRecordCompanyPSCApi;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.pscdataapi.data.IndividualPscRoles;
+import uk.gov.companieshouse.pscdataapi.data.SecurePscRoles;
+import uk.gov.companieshouse.pscdataapi.exceptions.FailedToTransformException;
+import uk.gov.companieshouse.pscdataapi.models.Created;
+import uk.gov.companieshouse.pscdataapi.models.NameElements;
+import uk.gov.companieshouse.pscdataapi.models.PscData;
+import uk.gov.companieshouse.pscdataapi.models.PscDocument;
+import uk.gov.companieshouse.pscdataapi.models.Updated;
+
+
+@Component
+public class CompanyPscTransformer {
+
+    private final DateTimeFormatter dateTimeFormatter =
+            DateTimeFormatter.ofPattern("yyyyMMddHHmmssSSSSSS");
+
+    private static final Logger logger = LoggerFactory.getLogger("psc-data-api");
+
+
+    /**
+     * Transform PSC.
+     * @param notificationId PSC Id.
+     * @param requestBody request payload.
+     * @return PSC mongo Document.
+     */
+    public PscDocument transformPsc(String notificationId, FullRecordCompanyPSCApi requestBody) {
+        PscDocument pscDocument = new PscDocument();
+
+        try {
+            pscDocument.setId(notificationId);
+            pscDocument.setNotificationId(notificationId);
+            pscDocument.setPscId(requestBody.getExternalData().getPscId());
+            pscDocument.setCompanyNumber(requestBody.getExternalData().getCompanyNumber());
+            OffsetDateTime deltaAt = requestBody.getInternalData().getDeltaAt();
+            pscDocument.setDeltaAt(dateTimeFormatter.format(deltaAt));
+            Created created = new Created();
```

Comment:
Could we split created and updated creations into their own method, just to get his method length down a bit